docs/index.html

```diff
@@ -1354,24 +1354,26 @@ <h3 id="name-semantics-6">
+<p id="section-2.5-5">Each matcher acts on its input list of nodes as follows. For each node in the list, the matcher may or may not match the node. If the matcher
                matches the node, it selects zero or more descendants of the node.
-               The output list of nodes of a matcher is the concatenation of the lists of selected descendents for each input node.<a href="#section-2.5-4" class="pilcrow">¶</a></p>
-<p id="section-2.5-5">A specific, non-normative example will make this clearer. Suppose the input document is: <code>{"a":[{"b":0},{"b":1},{"b":2}]}</code>. As we will see later, the JSON Path
-               <code>$.a[*].b</code> selects the following list of nodes: <code>0</code>, <code>1</code>, <code>2</code>. Let's walk through this in detail.<a href="#section-2.5-5" class="pilcrow">¶</a></p>
-<p id="section-2.5-6">The JSON Path consists of <code>$</code> followed by three matchers: <code>.a</code>, <code>[*]</code>, and <code>.b</code>.<a href="#section-2.5-6" class="pilcrow">¶</a></p>
-<p id="section-2.5-7">Firstly, <code>$</code> matches the input document and selects the root node which is the input document. So the result is a list consisting of just the root node.<a href="#section-2.5-7" class="pilcrow">¶</a></p>
-<p id="section-2.5-8">Next, <code>.a</code> matches any input node of type object and selects any value of the input node corresponding to the key <code>"a"</code>.
-               The result is again a list of one node: <code>[{"b":0},{"b":1},{"b":2}]</code>.<a href="#section-2.5-8" class="pilcrow">¶</a></p>
-<p id="section-2.5-9">Next, <code>[*]</code> matches any input node which is an array and selects all the elements of the input node.
-               The result is a list of three nodes: <code>{"b":0}</code>, <code>{"b":1}</code>, and <code>{"b":2}</code>.<a href="#section-2.5-9" class="pilcrow">¶</a></p>
-<p id="section-2.5-10">Finally, <code>.b</code> matches any input node of type object and selects any value of the input node corresponding to the key <code>"b"</code>.
-               The result is the concatenation of three lists each of length one containing <code>0</code>, <code>1</code>, <code>2</code>, respectively.<a href="#section-2.5-10" class="pilcrow">¶</a></p>
-<p id="section-2.5-11">As a consequence of this approach, if any of the matchers matches no nodes, then the whole JSON Path matches no nodes.<a href="#section-2.5-11" class="pilcrow">¶</a></p>
-<p id="section-2.5-12">In what follows, the semantics of each matcher are defined for each type of node.<a href="#section-2.5-12" class="pilcrow">¶</a></p>
+               The output list of nodes of a matcher is the concatenation of the lists of selected descendents for each input node.<a href="#section-2.5-5" class="pilcrow">¶</a></p>
+<p id="section-2.5-6">A specific, non-normative example will make this clearer. Suppose the input document is: <code>{"a":[{"b":0},{"b":1},{"b":2}]}</code>. As we will see later, the JSON Path
```

Comment:
Maybe we should define the last element of array as `{"c":2}` - just to demonstrate, how non-matching node is filtered out?

Reply:
Added.